src/aosm/azext_aosm/build_processors/base_processor.py

```diff
@@ -120,114 +120,175 @@ def generate_params_schema(self) -> Dict[str, Any]:
 
     def _generate_schema(
         self,
-        schema: Dict[str, Any],
+        deploy_params_schema: Dict[str, Any],
         source_schema: Dict[str, Any],
-        values: Dict[str, Any],
+        default_values: Dict[str, Any],
+        param_prefix: Optional[str] = None,
     ) -> None:
         """
-        Generate the parameter schema
+        Generate the JSON schema for deployParameters. The NFDProcessor overrides this method to generate CGS for NSDs.
 
-        This function recursively generates the parameter schema for the input artifact by updating
+        This method recursively generates the deployParameters schema for the input artifact by updating
         the schema parameter.
 
-        :param schema: The schema to generate.
-        :type schema: Dict[str, Any]
-        :param source_schema: The source schema.
-        :type source_schema: Dict[str, Any]
-        :param values: The values to generate the schema from.
-        :type values: Dict[str, Any]
+        Note there is no return value. The schema is passed by reference and modified in place.

```

Comment:
I'd like to see a comment here explaining how we treat each parameter type at the level of "add to schema with default/don't add to schema", etc. It's hard to work this out from the comments in the function body.

Reply:
Done